addons/api/addon/serializers/application.js

```diff
@@ -72,6 +73,20 @@ export default class ApplicationSerializer extends RESTSerializer {
       }
       delete json[key];
     }
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excluded
+    // from serialization.
```

Comment:
We should also probably be able to handle secrets but I don't think we need to worry about it yet, maybe just make a note in the comment that we don't handle it yet and can add support if/when we need it?